tests/chainer_tests/test_link.py

```diff
@@ -2466,4 +2466,43 @@ def test_to_intel64(self):
         assert l.child.to_method_called == 1
 
 
+class TestSerialize(unittest.TestCase):
+    def setUp(self):
+        self.array = numpy.array([1, 2, 3], dtype=numpy.float32)
+        self.serializer = mock.MagicMock(return_value=self.array)
+
+        # Create an empty link to populate.
+        link = chainer.Link()
+        with link.init_scope():
+            link.x = chainer.Parameter()
+            link.y = chainer.Parameter()
+        link.add_persistent('z', None)
+        self.link = link
+
+    def check_serialize(self, device_spec):
+        array = self.array
+        link = self.link
+        serializer = self.serializer
+
+        link.to_device(device_spec)
+        link.serialize(serializer)
+
+        self.assertEqual(serializer.call_count, 3)
+        cpu_device = chainer.backend.CpuDevice()
+        numpy.testing.assert_array_equal(cpu_device.send(link.x.array), array)
+        numpy.testing.assert_array_equal(cpu_device.send(link.y.array), array)
+        numpy.testing.assert_array_equal(cpu_device.send(link.z), array)
+
+    def test_serialize_numpy(self):
```

Comment:
Why not use `inject_backend_config`?

Reply:
Thanks, that actually helped me find a bug when the device is a ChainerX CUDA device. Fixed that bug and the test.